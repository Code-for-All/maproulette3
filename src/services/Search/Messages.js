import { defineMessages } from 'react-intl'

/**
 * Internationalized messages for use with Sort.
 */
export default defineMessages({
  name: {
    id: "Challenge.sort.name",
    defaultMessage: "Name",
  },
  created: {
    id: "Challenge.sort.created",
    defaultMessage: "Newest",
  },
  popularity: {
    id: "Challenge.sort.popularity",
    defaultMessage: "Popular",
  },
  default: {
    id: "Challenge.sort.default",
    defaultMessage: "Default",
  },
})
