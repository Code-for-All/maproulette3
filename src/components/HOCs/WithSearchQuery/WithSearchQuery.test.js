import React, { Component } from 'react'
import { _WithSearchQuery,
         mapStateToProps,
         mapDispatchToProps } from './WithSearchQuery'
import { setSearch, clearSearch } from '../../../services/Search/Search'

jest.mock('../../../services/Search/Search')

let basicState = null
let WrappedComponent = null

beforeEach(() => {
  basicState = {
    currentSearch: {
      searchQueryTest: "searchQueryValue",
    },
  }

  WrappedComponent = _WithSearchQuery(
    () => <div className="child" />,
    "searchQueryTest"
  )
})

test("searchQueries are passed through to the wrapped component", () => {
  const wrapper = shallow(
    <WrappedComponent {...basicState} />
  )

  expect(
    wrapper.props().searchQueries.searchQueryTest.searchQuery
  ).toBe(basicState.currentSearch.searchQueryTest)

  expect(wrapper).toMatchSnapshot()
})

test("mapStateToProps maps currentSearch", () => {
  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.currentSearch).toEqual(basicState.currentSearch)

  expect(mappedProps).toMatchSnapshot()
})

test("mapDispatchToProps maps function setSearch", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.setSearch("query", "searchName")
  expect(dispatch).toBeCalled()
  expect(setSearch).toBeCalledWith("searchName", "query")

  expect(mappedProps).toMatchSnapshot()
})

test("mapDispatchToProps maps function clearSearch", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.clearSearch("searchName")
  expect(dispatch).toBeCalled()
  expect(clearSearch).toBeCalledWith("searchName")
})
