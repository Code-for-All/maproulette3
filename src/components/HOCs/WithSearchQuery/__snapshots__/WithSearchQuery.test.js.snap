// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDispatchToProps maps function setSearch 1`] = `
Object {
  "clearSearch": [Function],
  "setSearch": [Function],
}
`;

exports[`mapStateToProps maps currentSearch 1`] = `
Object {
  "currentSearch": Object {
    "searchQueryTest": "searchQueryValue",
  },
}
`;

exports[`searchQueries are passed through to the wrapped component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    currentSearch={
        Object {
            "searchQueryTest": "searchQueryValue",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "searchQueries": Object {
        "searchQueryTest": Object {
          "clearSearch": [Function],
          "searchQuery": "searchQueryValue",
          "setSearch": [Function],
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "searchQueries": Object {
          "searchQueryTest": Object {
            "clearSearch": [Function],
            "searchQuery": "searchQueryValue",
            "setSearch": [Function],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
